### PR TITLE
Improve base templates

### PR DESCRIPTION
**Pull request by ruflin**
This PR is to iterated on the base templates to come to a conclusion what the content should be.

At first, trying out how our validation will behave if I use comments in the template (which are valid in ES).

### CODE REVIEW

File: dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json [hunk 1]
```diff
@@ -5,15 +5,16 @@
                 "lifecycle": {
                     "name": "logs-default"
                 },
+                // Best compression optimises for storage
```
Comment (ruflin) [inline dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json:8]: @jpountz This works with ES but most validators and Github are not very happy with this. What does ES use to validate these?

**Comment by ruflin** [comment]
I realised perhaps a better way to write our index template and component templates is in yaml as it also allows to add comments and works around the issue of having comments in json. In addition, we already use yaml for quite a few ingest pipelines and often my personal view is, it makes them more readable. I think the same applies here.

@skh Would it be possible to support also yaml index templates and component templates in EPM?

File: dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json (hunk 1 above)
Comment (jpountz) [reply dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json:8]: Do you mean validating JSON files that include comments? We're relying on the library we use for JSON parsing, Jackson. https://github.com/FasterXML/jackson

Comment (ruflin) [reply dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json:8]: Yes, exactly. Most JSON validators I have seen so complain if they see comments. But potentially my workaround is to just use yaml instead (see other change in this PR).

**Comment by skh** [comment]
> Would it be possible to support also yaml index templates and component templates in EPM?

As long as ES accepts them in yaml (I haven't tested that), we handle it the same way as we do with ingest pipelines and decide based on the file ending what format to send, see https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/ingest_pipeline/install.ts#L141-L143 .

So, yes, but it needs a code change in Ingest Manager / EPM.

**Comment by ruflin** [comment]
@skh Yes, ES accepts yaml, tested it on my end.
```
curl -X PUT "elastic:changeme@localhost:9200/_component_template/foo?pretty" -H 'Content-Type: application/yaml' -d'
template:
  settings:
    index:
      codec: "best_compression"
      refresh_interval: "5s"
      query:
        default_field:
          - "message"
'
```
I will file an issue in KB for it.

**Comment by ruflin** [comment]
Issue in Kibana repo: https://github.com/elastic/kibana/issues/67318

File: dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json (hunk 1 above)
Comment (jpountz) [reply dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json:8]: I'll bring this question to our Fix-it meeting to discuss what should be the way forward.

Comment (ruflin) [reply dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json:8]: @jpountz Not sure if ES needs to fix anything here as in ES things work just as expected (as long as I don't try to validate it somewhere :-) ).

Comment (jpountz) [reply dev/packages/example/base/0.2.0/elasticsearch/component-template/logs-settings.json:8]: Maybe not indeed, but I think it's important to at least raise awareness about this problem, ie. the fact that we allow comments in JSON doesn't mean that it's easy for consumers of the Elasticsearch APIs to leverage this feature.

**Comment by ruflin** [comment]
Closing this PR as changes to the base templates now happen in https://github.com/elastic/elasticsearch/pull/57629 As soon as this is merged, we should get the changes into the integrations repo.

**Pull request closed without merging**